src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -456,89 +366,45 @@ bool Profile::_ConvertJsonToBool(const Json::Value& json)
 void Profile::LayerJson(const Json::Value& json)
 {
     // Profile-specific Settings
-    JsonUtils::GetWstring(json, NameKey, _name);
-
-    JsonUtils::GetOptionalGuid(json, GuidKey, _guid);
-
-    JsonUtils::GetBool(json, HiddenKey, _hidden);
+    JsonUtils::GetValueForKey(json, NameKey, _name);
+    JsonUtils::GetValueForKey(json, GuidKey, _guid);
+    JsonUtils::GetValueForKey(json, HiddenKey, _hidden);
 
     // Core Settings
-    JsonUtils::GetOptionalColor(json, ForegroundKey, _defaultForeground);
-
-    JsonUtils::GetOptionalColor(json, BackgroundKey, _defaultBackground);
-
-    JsonUtils::GetOptionalColor(json, SelectionBackgroundKey, _selectionBackground);
-
-    JsonUtils::GetOptionalColor(json, CursorColorKey, _cursorColor);
-
-    JsonUtils::GetOptionalString(json, ColorSchemeKey, _schemeName);
+    JsonUtils::GetValueForKey(json, ForegroundKey, _defaultForeground);
+    JsonUtils::GetValueForKey(json, BackgroundKey, _defaultBackground);
+    JsonUtils::GetValueForKey(json, SelectionBackgroundKey, _selectionBackground);
+    JsonUtils::GetValueForKey(json, CursorColorKey, _cursorColor);
+    JsonUtils::GetValueForKey(json, ColorSchemeKey, _schemeName);
 
     // TODO:MSFT:20642297 - Use a sentinel value (-1) for "Infinite scrollback"
-    JsonUtils::GetInt(json, HistorySizeKey, _historySize);
-
-    JsonUtils::GetBool(json, SnapOnInputKey, _snapOnInput);
-
-    JsonUtils::GetBool(json, AltGrAliasingKey, _altGrAliasing);
-
-    JsonUtils::GetUInt(json, CursorHeightKey, _cursorHeight);
-
-    if (json.isMember(JsonKey(CursorShapeKey)))
-    {
-        auto cursorShape{ json[JsonKey(CursorShapeKey)] };
-        _cursorShape = _ParseCursorShape(GetWstringFromJson(cursorShape));
-    }
-    JsonUtils::GetOptionalString(json, TabTitleKey, _tabTitle);
+    JsonUtils::GetValueForKey(json, HistorySizeKey, _historySize);

```

Comment:
idea2:

```
JsonUtils::Context parsingContext{ json };
parsingContext.GetValueForKey(Key, _v);
parsingContext.GetValueForKey(Key, _v);
parsingContext.GetValueForKey(Key, _v);
```

also, `parsingContext` could accumulate errors and exceptions, instead of throwing them on-the-spot.

Reply:
I love this, but feel free to make that a follow-up